app/presenters/error_presenter.rb

```diff
@@ -21,38 +21,25 @@ def initialize(claim, message_file = nil)
   def generate_messages
     @errors.each do |error|
       attribute = error.attribute
-      messages = populate_messages(error)
-      next if @error_details[attribute] && @error_details[attribute][0].long_message.eql?(messages.long)
+      messages = translate(error)
+      next if @error_details[attribute] && @error_details[attribute][0].long_message.eql?(messages.long_message)
+
       @error_details[attribute] = ErrorDetail.new(
         attribute,
-        messages.long,
-        messages.short,
-        messages.api,
+        messages.long_message,
+        messages.short_message,
+        messages.api_message,
         generate_sequence(attribute)
       )
     end
   end
 
-  def populate_messages(error)
-    emt = ErrorMessageTranslator.new(@translations, error.attribute, error.message)
-
-    if emt.translation_found?
-      OpenStruct.new(
-        long: emt.long_message,
-        short: emt.short_message,
-        api: emt.api_message
-      )
-    else
-      OpenStruct.new(
-        long: generate_standard_long_message(error),
-        short: generate_standard_short_message(error),
-        api: generate_standard_api_message(error)
-      )
-    end
+  def translate(error)
+    ErrorMessage::Translator.new(@translations, error.attribute, error.message)
```

Comment:
@translator = ErrorMessage::Translator.new(@translations)

@translator.translate(error)

ErrorMessage::Translator.new(error.attribute, error.message)

Reply:
see https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/4180